### DISCUSSION TITLE
Incorrect openapi.json when using multiple APIKeyHeaders with different names

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#6335] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Security
from fastapi.security import api_key

app = FastAPI()

api_key_header_a = api_key.APIKeyHeader(name='key_a')
api_key_header_b = api_key.APIKeyHeader(name='key_b')


@app.get("/a")
async def a(auth_token: str = Security(api_key_header_a)):
    pass


@app.get('/b')
async def b(auth_token: str = Security(api_key_header_b)):
    pass
```

### Description

- Start the app with `uvicorn main:app`
- Go to http://127.0.0.1:8000/openapi.json
- Note that the specification only contains one of the two api key headers
- Note that http://127.0.0.1:8000/docs hence doesn't work properly: any authorization sets the header 'key_b'

I would expect that the openapi specification contains both api key headers, and that that in the Swagger UI, clicking the lock on endpoint 'a' sets header 'key_a' and the lock on endpoint 'b' sets header 'key_b'.

I also want to mention that the code itself works fine, only the openapi specification is incorrect.

### Operating System

Linux

### FastAPI Version

0.73.0

### Python Version

3.9.10

### Additional Context

![screenshot_2022-03-02-115416](https://user-images.githubusercontent.com/26491995/156348396-d1d3d7af-b209-480d-9f1c-129ef623c721.png)
As we can see, the Swagger UI wants to authorize against endpoint 'a' by setting 'key_b', which is not what is defined in the code.

**Comment by BilalAlpaslan** [comment]
you should set schema name for openapi
```py
from fastapi import FastAPI, Security
from fastapi.security import api_key

app = FastAPI()

api_key_header_a = api_key.APIKeyHeader(name='key_a', scheme_name='api_key_header_a')
api_key_header_b = api_key.APIKeyHeader(name='key_b', scheme_name='api_key_header_b')


@app.get("/a")
async def a(auth_token: str = Security(api_key_header_a, use_cache=False)):
    pass


@app.get('/b')
async def b(auth_token: str = Security(api_key_header_b, use_cache=False)):
    pass
```

**Reply by BilalAlpaslan** [reply]
permanent solution: https://github.com/tiangolo/fastapi/pull/4791
